src/EnergyPlus/ChillerAbsorption.cc

```diff
@@ -456,22 +456,17 @@ void GetBLASTAbsorberInput(EnergyPlusData &state)
         thisChiller.TempLowLimitEvapOut = state.dataIPShortCut->rNumericArgs(15);
 
         {
-            auto const SELECT_CASE_var(state.dataIPShortCut->cAlphaArgs(8));
-            if (SELECT_CASE_var == "CONSTANTFLOW") {
-                thisChiller.FlowMode = DataPlant::FlowMode::Constant;
-            } else if (SELECT_CASE_var == "LEAVINGSETPOINTMODULATED") {
-                thisChiller.FlowMode = DataPlant::FlowMode::LeavingSetpointModulated;
-            } else if (SELECT_CASE_var == "NOTMODULATED") {
-                thisChiller.FlowMode = DataPlant::FlowMode::NotModulated;
-            } else {
+            thisChiller.FlowMode =
+                static_cast<DataPlant::FlowMode>(getEnumerationValue(DataPlant::FlowModeNamesUC, state.dataIPShortCut->cAlphaArgs(8)));
+            if (thisChiller.FlowMode == DataPlant::FlowMode::Invalid) {
```

Comment:
So, we are not doing these specific checks anymore because the epJSON parser does them already, but I wonder what the situation is in this case where in case of `Invalid` there is a default mode. @rraustad? @mjwitte? @Myoldmopar?

Reply:
Yeah, I thought of that. If this warning is after the input read (state.dataIPShortCut->cAlphaArgs(8)), then that read must have been a valid input. Am I missing something else here not local to these lines of code?

    Chiller:Absorption:Indirect,
     A6 , \field Chiller Flow Mode
           \type choice
           \key ConstantFlow
           \key LeavingSetpointModulated
           \key NotModulated
           \default NotModulated

Reply:
Maybe there needs to be another rule that the field in question needs to be before or at min fields. At least when using getObjectItem. I think @mjwitte has indicated that json will fill these but I am not sure of that point.

    Chiller:Absorption:Indirect,
       \min-fields 17

Reply:
1. If a default is declared in the IDD, then GetObjectItem will fill the default value into a blank alpha or numeric field *if* the field is within \min-fields. If epJSON is accessed directly, then the field will come back blank. 
2. If epJSON is accessed through `GetAlphaField` or `getRealField` the default will be filled in. 
3. So, there are two ways a blank could get passed to `getEnumerationValue` even if there's a default in the IDD. Most current `getInput` functions check for blank fields and assign the default or initialize the struct to the default value. 
4. This isn't perfect, because the IDD default could be inconsistent with the code default, but this is present all over the code now. 
5. For the case of choice fields, would it be a good idea to add an optional default value argument that is returned if the incoming alpha is blank? This has the same possible problem of being inconsistent with the IDD default, but it would show the default explicitly in the `getInput` function (which I think is useful when reading the code, you don't have to go find the input field in the IDD to find out it's default).